charts/seed-monitoring/charts/core/charts/prometheus/templates/config-konnectivity.yaml

```diff
@@ -0,0 +1,720 @@
+{{- if .Values.konnectivityTunnel.enabled}}
```

Comment:
Hm, without having looked into it with more detail it seems that most part of this prometheus config are the same. Can you keep one file and one enable/disable the relevant rules either for the konnectivity tunnel or for VPN? Now we have two files that are almost identical.

Reply:
Not really, I am using proxy instead of vpn in almost every job (which means different config). Having one file will lead to many if conditions.

Reply:
Can you show what exactly is different in the config when you are using proxy instead of VPN? Probably this can be extracted in a template and then included at the relevant places? I'd really try to avoid having this large file twice.

Reply:
As an example, have a look at this: 

```yaml
    - job_name: kube-kubelet
      honor_labels: false
      scheme: https
      tls_config:
        # This is needed because the kubelets' certificates are not are generated
        # for a specific pod IP
        insecure_skip_verify: true
        cert_file: /srv/kubernetes/prometheus-kubelet/prometheus-kubelet.crt
        key_file: /srv/kubernetes/prometheus-kubelet/prometheus-kubelet.key
      kubernetes_sd_configs:
      - role: node
        api_server: https://kube-apiserver:443
        tls_config:
{{ include "prometheus.tls-config.kube-cert-auth" . | indent 10 }}
      relabel_configs:
      - source_labels: [ __meta_kubernetes_node_address_InternalIP ]
        target_label: instance
      - action: labelmap
        regex: __meta_kubernetes_node_label_(.+)
      - target_label: type
        replacement: shoot
``` 
Vs. this: 

```
    - job_name: kube-kubelet
      honor_labels: false
      scheme: https
      tls_config:
        # This is needed because the kubelets' certificates are not are generated
        # for a specific pod IP
        insecure_skip_verify: true
        cert_file: /etc/prometheus/seed/prometheus.crt
        key_file: /etc/prometheus/seed/prometheus.key
      kubernetes_sd_configs:
      - role: node
        api_server: https://kube-apiserver:443
        tls_config:
{{ include "prometheus.tls-config.kube-cert-auth" . | indent 10 }}
      relabel_configs:
      - source_labels: [ __meta_kubernetes_node_address_InternalIP ]
        target_label: instance
      - action: labelmap
        regex: __meta_kubernetes_node_label_(.+)
      - target_label: __address__
        replacement: kube-apiserver:443
      - source_labels: [__meta_kubernetes_node_name]
        regex: (.+)
        target_label: __metrics_path__
        replacement: /api/v1/nodes/${1}:10250/proxy/metrics
      - target_label: type
        replacement: shoot
```


Other different configuration occur for `kube-proxy, coredns, apiservice check, cadvisor,..`

I can try to templatize some common parts but still keep them as two separate files otherwise it gets confusing in one file.

Reply:
OK, I see, thanks. I don't understand most of these configurations, so if it's too hard to move it into a template then that's how it is. @wyb1 or @tim-ebert can judge way better than myself, what are your thoughts?

Reply:
we can also use proxy all the way and remove the VPN pod from prometheus in both cases, its only really needed for the API server when VPN is enabled, this way we do the check similarly in both cases and have one file, what do you think @rfranzke @wyb1?

Reply:
Sure, sounds reasonable 👍 @wyb1 is there anything speaking against it?

Reply:
I am ok with this as long as we are not removing any prometheus jobs